.github/workflows/release-plz.yml

```diff
@@ -12,32 +12,30 @@ jobs:
     if: ${{ github.repository_owner == 'release-plz' }}
     environment: release-plz
     permissions:
-      contents: write
+      contents: read
       # Required for trusted publishing
       id-token: write
     steps:
-      # Generating a GitHub token, so that PRs and tags created by
-      # the release-plz-action can trigger actions workflows.
-      - name: Generate GitHub token
-        uses: actions/create-github-app-token@67018539274d69449ef7c02e8e71183d1719ab42 # v2.1.4
-        id: generate-token
-        with:
-          app-id: ${{ secrets.APP_ID }}
-          private-key: ${{ secrets.APP_PRIVATE_KEY }}
       - &checkout
         name: Checkout repository
         uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
         with:
           fetch-depth: 0
-          token: ${{ steps.generate-token.outputs.token }}
-          # Allow the release-plz action to push.
-          # `true` is the default value.
-          persist-credentials: true
+          persist-credentials: false
       - &setup-rust
         name: Install Rust toolchain
         uses: dtolnay/rust-toolchain@e97e2d8cc328f1b50210efc529dca0028893a2d9
         with:
           toolchain: stable
+      # Generating a GitHub token, so that PRs and tags created by
+      # the release-plz-action can trigger actions workflows.
+      - name: Generate GitHub token
+        uses: actions/create-github-app-token@67018539274d69449ef7c02e8e71183d1719ab42 # v2.1.4
+        id: generate-token
+        with:
+          app-id: ${{ secrets.APP_ID }}
+          private-key: ${{ secrets.APP_PRIVATE_KEY }}
+          permission-contents: write
```

Comment:
why is this required now if before wasn't?
The permissions of the app are managed in the app settings anyway, why do we need to repeat them here? 🤔